internal/pkg/skuba/deployments/ssh/ssh.go

```diff
@@ -169,10 +174,17 @@ func (t *Target) internalSshWithStdin(silent bool, stdin string, command string,
 	if err := session.Start(finalCommand); err != nil {
 		return "", "", err
 	}
+
+	spin := spinner.New(spinner.CharSets[36], 1000*time.Millisecond)
+	if !t.no_progress {
+		spin.Start()
+		defer spin.Stop()
+	}
 	stdoutChan := make(chan string)
 	stderrChan := make(chan string)
-	go readerStreamer(stdoutReader, stdoutChan, "stdout", silent)
-	go readerStreamer(stderrReader, stderrChan, "stderr", silent)
+
+	go readerStreamer(stdoutReader, stdoutChan, "stdout", silent, spin, t.no_progress)
+	go readerStreamer(stderrReader, stderrChan, "stderr", silent, spin, t.no_progress)
```

Comment:
This is probably the right place to add the tty check.  If stdout or stderr are not connected to TTYs, just pass in false always, ignoring t.no_progress.  Maybe something like "isatty(sys.stderr) || t.no_progress", or however that would be properly written in Go. :D

Reply:
@dannysauer Thanks for the different viewpoint, I will look into it. If this works, then I do not need to have the `--no-progress` option, which is even greater.